drivers/QSPI.h

```diff
@@ -160,7 +166,7 @@ class QSPI : private NonCopyable<QSPI> {
      *  @returns
      *    Returns QSPI_STATUS_SUCCESS on successful reads and QSPI_STATUS_ERROR on failed reads.
      */
-    qspi_status_t read(int instruction, int alt, int address, char *rx_buffer, size_t *rx_length);
+    qspi_status_t read(qspi_inst_t instruction, int alt, int address, char *rx_buffer, size_t *rx_length);
```

Comment:
`int` should be sufficient for inst as it can be any number, narrow down this won't make much difference here

Reply:
The usage of this in the block device was wrong, should follow what the driver specifies.

Reply:
> The usage of this in the block device was wrong, should follow what the driver specifies.
Can you clarify what you mean? Are you saying that the use of `unsigned int` instead of `int` in the block device was wrong?

Reply:
> `int` should be sufficient for inst as it can be any number, narrow down this won't make much difference here.
> 
> Same as alt, address and others.
> 
> This is breaking change, so needs to be justified to go in.

This change is not mandatory for the fix here so I can pull it out of this review and into a separate PR if it makes things easier. But in principle, shouldn't values which correspond to bit patterns of a particular width be represented as types of particular widths? If so, then I would argue that a separate PR should also replace  `alt` and `address` with more specific types.

Reply:
I'm not that familiar with SQPIF, but if commands are limited to 8 bytes, then I would say that this is acceptable, and it is not a breaking change.

Mbed OS have not guaranteed to have binary compatibility, so moving from int to uint8_t should be OK.